Open Judge System/Web/OJS.Web/Areas/Contests/ViewModels/Results/ParticipantResultViewModel.cs

```diff
@@ -31,42 +31,5 @@ public class ParticipantResultViewModel
             .Sum(pr => pr.BestSubmission.Points);
 
         public IEnumerable<int> ParticipantProblemIds { get; set; }
-
-        public static Expression<Func<Participant, ParticipantResultViewModel>> FromParticipantAsSimpleResultByContest(int contestId) =>
-            participant => new ParticipantResultViewModel
-            {
-                ParticipantUsername = participant.User.UserName,
-                ParticipantFirstName = participant.User.UserSettings.FirstName,
-                ParticipantLastName = participant.User.UserSettings.LastName,
-                ParticipantProblemIds = participant.Problems.Select(p => p.Id),
-                ProblemResults = participant.Scores
-                    .Where(sc => !sc.Problem.IsDeleted && sc.Problem.ProblemGroup.ContestId == contestId)
-                    .AsQueryable()
-                    .Select(ProblemResultPairViewModel.FromParticipantScoreAsSimpleResult)
-            };
-
-        public static Expression<Func<Participant, ParticipantResultViewModel>> FromParticipantAsFullResultByContest(int contestId) =>
-            participant => new ParticipantResultViewModel
-            {
-                ParticipantUsername = participant.User.UserName,
-                ParticipantFirstName = participant.User.UserSettings.FirstName,
-                ParticipantLastName = participant.User.UserSettings.LastName,
-                ProblemResults = participant.Scores
-                    .Where(sc => !sc.Problem.IsDeleted && sc.Problem.ProblemGroup.ContestId == contestId)
-                    .AsQueryable()
-                    .Select(ProblemResultPairViewModel.FromParticipantScoreAsFullResult)
-            };
```

Comment:
Why did you remove those? I see that this logic is moved inside the controller. Can't you leave it here (with small refactoring of the queries if needed)

Reply:
![image](https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/77450034/d8d10f74-58f6-4453-8d90-65cadb2500e3)
 It was moved inside the controller because of this error.

Reply:
This logic was moved to the controller because of this error: The type arguments cannot be inferred from the usage. Try specifying the type arguments explicitly. Candidates are:   System.Collections.Generic.IEnumerable<TResult> Select<Participant,TResult>(this System.Collections.Generic.IEnumerable<OJS.Data.Models.Participant>, System.Func<OJS.Data.Models.Participant,int,TResult>) (in class Enumerable)   System.Collections.Generic.IEnumerable<TResult> Select<Participant,TResult>(this System.Collections.Generic.IEnumerable<OJS.Data.Models.Participant>, System.Func<OJS.Data.Models.Participant,TResult>) (in class Enumerable)

Reply:
You can fix this 💯 
Try `.AsQueryable`

Reply:
![image](https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/77450034/bb6650e6-7360-494b-b8c8-7939293f1bde)
 It is not working, it shows the same error

Reply:
`.AsQueryable` should be called before the `.Select`